Use empty object status list when the StatusPolicyNone is used

Liujingfang1 opened the pull request:
This reduces the request size when StatusPolicyNone is used

karlkfi commented on cmd/main.go:
```diff
@@ -56,7 +56,7 @@ func main() {
 	updateHelp(names, initCmd)
 	loader := manifestreader.NewManifestLoader(f)
-	invFactory := inventory.ClusterClientFactory{}
+	invFactory := inventory.ClusterClientFactory{StatusPolicy: inventory.StatusPolicyAll}
```
do we need this if the default ConfigMap inventory doesn't store the status anyway? That's why i had left it to default to None before.

Liujingfang1 replied:
We injected the apply status into the ConfigMap inventory. Updated the InventoryCM, now we don't need to change this to All.

karlkfi reviewed (changes requested):
The default inventory provider was deliberately StatusPolicyNone, because the custom inventory provider tests StatusPolicyAll. Why change it?

You might want to add some unit tests that test a custom provider so that we know the status update works when StatusPolicyAll and doesn't write them when StatusPolicyNone.

Liujingfang1 commented:
> The default inventory provider was deliberately StatusPolicyNone, because the custom inventory provider tests StatusPolicyAll. Why change it?

We injected the apply status into the inventory configmap. If the passed in `[]ObjectStatus` is empty, it throws an error. Updated the code to allow this.

> You might want to add some unit tests that test a custom provider so that we know the status update works when StatusPolicyAll and doesn't write them when StatusPolicyNone.

Added an updated unit tests for this.

karlkfi reviewed (approved):
/lgtm
